skimage/draw/draw3d.py

```diff
@@ -63,6 +64,165 @@ def ellipsoid(a, b, c, spacing=(1., 1., 1.), levelset=False):
+        produce the rotations are specified by ``axes``. No rotation will be
+        applied if ``angles`` is ``None``.
+    axes : iterable of int, length 3, optional (default None)
+        Axes about which to produce the rotations.
+        Defaults to ``[0, 1, 2]``.
+    intrinsic : bool, optional
+        Specify if the rotations are intrinsic (``True``) or extrinsic
+        (``False``).
+
+        - *intrinsic*: rotations occur about the rotating coordinate system
+        - *extrinsic*: rotations occur about the original coordinate system
+    spacing : (3,) array-like of float, optional
+        Spacing in each spatial dimension.  Defaults to 1.
+
+    Returns
+    -------
+    dd, rr, cc : ndarray of int
+        Voxel coordinates of ellipsoid.
+        May be used to index into an array, e.g.
+        ``img[dd, rr, cc] = 1``
+
+    Notes
+    ------
+    ``angles`` are defined as the Euler angles based on ``axes`` and
+    ``intrinsic`` parameters. The following example shows the ``0-2-0``
+    (``z-x-z``) intrinsic rotations.
+
+    .. math::
+
+        R(\phi) = \begin{bmatrix}
+                      1 &        0 &         0 \\
+                      0 & \cos\phi & -\sin\phi \\
+                      0 & \sin\phi &  \cos\phi \\
+                  \end{bmatrix}
+
+        R(\theta) = \begin{bmatrix}
+                        \cos\theta & -\sin\theta & 0 \\
+                        \sin\theta &  \cos\theta & 0 \\
+                                 0 &           0 & 1 \\
+                    \end{bmatrix}
+
+        R(\psi) = \begin{bmatrix}
+                      1 &        0 &         0 \\
+                      0 & \cos\psi & -\sin\psi \\
+                      0 & \sin\psi &  \cos\psi \\
+                           \end{bmatrix}
+
+        \\
+
+        R(\phi, \theta, \psi) = (R(\psi)R(\theta)R(\phi))^\intercal
+
+    """
+
+    if len(center) != 3:
+        raise ValueError(f'len(center) should be 3 but got {len(center)}')
+    center = np.array(center)
+
+    if len(axis_lengths) != 3:
+        raise ValueError(
+            f'len(axis_lengths) should be 3 but got {len(axis_lengths)}')
+    axis_lengths = np.array(axis_lengths)
+
+    if angles is None:
+        R = np.eye(3)
+    else:
+        if len(angles) != 3:
+            raise ValueError(
+                'len(angles) should be 3 but got',
+                len(angles))
+        if axes is None:
+            axes = range(3)
+        if (len(axes) != 3
+            or not set(axes).issubset(range(3))
+            or axes[0] == axes[1]
+                or axes[1] == axes[2]):
+            raise ValueError(f'axes: {axes} is invalid')
+
+        axes_str = 'ZYX' if intrinsic else 'zyx'
+        seq = ''.join(axes_str[axis] for axis in axes)
+
+        # Generate a rotation matrix. The order of the elements needs to be
+        # reversed so that it follows the (z, y, x) order.
+        R = Rotation.from_euler(seq, angles).as_matrix()[::-1, ::-1]
```

Comment:
What is this inversion doing exactly? And why is `axes_str` defined as `ZYX` instead of `XYZ`? 
Are there two definitions just inverting one another?

Reply:
Typically, axis 0 corresponds not to X but to Z. Think, e.g., of an image represented as an array: `(M, N)` means `(Y, X)`, upside down, sort of. You see the confusion :)

Reply:
I see, this is because of the application for images then!